### PR TITLE
Headers clarifications

### PR DESCRIPTION
**Pull request by liamdennehy**
In response to https://github.com/w3c-dvcg/http-signatures/issues/41, https://github.com/w3c-dvcg/http-signatures/issues/32, https://github.com/w3c-dvcg/http-signatures/issues/42, hopefully this PR can clarify how to handle weird or bad headers parameters:

* Explicit handling of empty headers parameter
* Explicit handling of empty header value in a message
* Clarification on case of headers parameter values (SHOULD, but not mandatory)
* Explicit rejection of signature generation for an unmatched header
* small typo

### CODE REVIEW

**Comment by liamdennehy** [comment]
This PR is aimed at resolving the ambiguities @lukasz-szewc pointed out in #41 and #42. The Signature String Construction section needs an overhaul as the flow is not obvious, especially if we introduce the response status from #27 - it should read more like a verbal state diagram.

Once the clarifications in this PR are approved I'll work on implementing a solution to #27 in a more logical flow.

**Review by msporny — APPROVED** [review]
LGTM. The fact that the base example is getting more complicated is probably not good. We need a simple example and then a corner-case example. The alternative is to start making progress on the test suite and move the complex examples out to the test suite.

Ready to merge when you are... is this a WIP or done?

**Comment by lukasz-szewc** [comment]
@liamdennehy @msporny very good - this PR clarifies concerns I raised inside #41 and #42

**Comment by liamdennehy** [comment]
@msporny 
> Ready to merge when you are... is this a WIP or done?

Maybe another couple of days to make sure it gets a chance for a review - especially from implementers

**Comment by liamdennehy** [comment]
@msporny 
> move the complex examples out to the test suite

Good point, I'll try to incorporate this in #43. Do you want to keep these here in the meantime?

**Comment by liamdennehy** [comment]
@msporny The PR has been reviewed by the related issue owner, and allowed to stew for a few days, I think we're good to go. Thx.

**Pull request merged**
